### PR TITLE
feat: add get all data

### PR DESCRIPTION
**Pull request by salesh**
Related to https://github.com/fastify/fastify-secure-session/issues/118

If you have any ideas on how to display this one in documentation (I couldn't figure out where to place it..)

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -240,6 +240,11 @@ class Session {
   options (opts) {
     this[kCookieOptions] = opts
   }
+
+  data () {
+    const { changed, deleted, ...data } = this[kObj]
+    return data
```
Comment (salesh) [inline index.js:246]: [Q] I thought we would want to exclude `changed` and `deleted`, what do you think?

Comment (Eomm) [reply index.js:246]: I would keep them outside and, if needed, add a new getter:

`dataChanged() {}` and `dataDeleted()`

**Review by Eomm — APPROVED** [review]

**Review by Eomm — COMMENTED** [review]
It miss docs

File: index.js (hunk 1 above)
Comment (salesh) [reply index.js:246]: If I am not wrong since we can `request.session.changed` and `request.session.deleted` there is no need for those two?

**Comment by salesh** [comment]
> It miss docs

Yes, I mention in the main comment that I'm having trouble identifying where I should add it - maybe a new section?

**Comment by Eomm** [comment]
> Yes, I mention in the main comment that I'm having trouble identifying where I should add it - maybe a new section?

I would go in `Using a secret` adding a new route to the example.

File: index.js (hunk 1 above)
Comment (Eomm) [reply index.js:246]: me neither, could you add a test for it and check if it works as expected?

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Eomm — APPROVED** [review]

**Pull request merged**